analysis/taint/taint_utils_test.go

```diff
@@ -253,40 +259,40 @@ func runTest(t *testing.T, dirName string, files []string, summarizeOnDemand boo
 		t.Fatal("no taint flows to sinks found")
 	}
 
-	astFs := analysistest.AstFiles(lp.Pkgs)
-	expectSinkToSources, expectEscapeToSources := expectedTaintTargetToSources(lp.Prog.Fset, astFs)
+	astFs := analysistest.AstFiles(lp.Packages)
+	expectSinkToSources, expectEscapeToSources := expectedTaintTargetToSources(lp.Program.Fset, astFs)
 
 	if len(expectSinkToSources) == 0 {
 		t.Fatal("no expected taint flows found")
 	}
 
-	checkExpectedPositions(t, lp.Prog, result.TaintFlows, expectSinkToSources, expectEscapeToSources)
+	checkExpectedPositions(t, lp.Program, result.TaintFlows, expectSinkToSources, expectEscapeToSources)
 	// Remove reports - comment if you want to inspect
 	os.RemoveAll(lp.Config.ReportsDir)
 }
 
 type runTestResult struct {
-	lp  analysistest.LoadedTestProgram
+	lp  resultMonad.Result[loadprogram.State]
 	res taint.AnalysisResult
 }
 
 // runTestWithoutCheck runs the test without checking expected flows.
 func runTestWithoutCheck(t *testing.T, dirName string, files []string, summarizeOnDemand bool, errorExpected func(e error) bool) runTestResult {
 	dirName = filepath.Join("./testdata", dirName)
-	lp, err := analysistest.LoadTest(testfsys, dirName, files, analysistest.LoadTestOptions{ApplyRewrite: false})
-	if err != nil {
-		t.Fatalf("failed to load test: %v", err)
+	lp := analysistest.LoadTest(testfsys, dirName, files, analysistest.LoadTestOptions{ApplyRewrite: false})
+	if lp.IsErr() {
+		t.Fatalf("failed to load test: %v", lp)
 	}
-	setupConfig(lp.Config, summarizeOnDemand)
-	state, err := dataflow.NewInitializedAnalyzerState(lp.Prog, lp.Pkgs, config.NewLogGroup(lp.Config), lp.Config)
+	resultMonad.Do(lp, func(lp *loadprogram.State) { setupConfig(lp.Config, summarizeOnDemand) })
+	ptrState := resultMonad.Bind(lp, ptr.NewState)
+	state, err := resultMonad.Bind(ptrState, dataflow.NewState).Value()
 	if err != nil {
 		t.Fatalf("failed to initialize state")
 	}
 	result, err := taint.Analyze(state)
```

Comment:
nit: could rename `result` here to `res` to avoid the `resultMonad` named import